sdk/core/core-util/src/delay.ts

```diff
@@ -28,38 +27,24 @@ export interface DelayOptions {
  */
 export function delay(timeInMs: number, options?: DelayOptions): Promise<void> {
   return new Promise((resolve, reject) => {
-    let timer: ReturnType<typeof setTimeout> | undefined = undefined;
-    let onAborted: (() => void) | undefined = undefined;
-
     const rejectOnAbort = (): void => {
```

Comment:
since we're cleaning this up, it might be nicer to have these simply be named functions instead of anonymous ones bound to a local const:

```ts
function rejectOnAbort() {
```